src/resolve/resolved-manager.c

```diff
@@ -1200,7 +1200,7 @@ int manager_compile_dns_servers(Manager *m, OrderedSet **dns) {
         return 0;
 }
 
-int manager_compile_search_domains(Manager *m, OrderedSet **domains, int filter_route) {
+int manager_compile_search_domains(Manager *m, OrderedSet **domains, bool filter_route) {
```

Comment:
hmm, the idea is that "filter route" would be a tri-state: negative (for "no filtering"), 0 (for "only routes") or positive (for "no routes"). We don't use it that way, but I still like this scheme conceptually I must say...